provider/plugin/httpapi.go

```diff
@@ -0,0 +1,168 @@
+}
+
+type PropertyValuesEqualsResponse struct {
+	Equals bool `json:"equals"`
+}
+
+func (p *HTTPProvider) recordsHandler(w http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case http.MethodGet:
+		records, err := p.provider.Records(context.Background())
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		json.NewEncoder(w).Encode(records)
+		return
+	case http.MethodPost:
+		var changes plan.Changes
+		if err := json.NewDecoder(req.Body).Decode(&changes); err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		err := p.provider.ApplyChanges(context.Background(), &changes)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		return
+	default:
+		log.Errorf("Unsupported method %s", req.Method)
+		w.WriteHeader(http.StatusBadRequest)
+	}
+}
+
+func (p *HTTPProvider) propertyValuesEqualHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		log.Errorf("Unsupported method %s", req.Method)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	pve := PropertyValuesEqualsRequest{}
+	if err := json.NewDecoder(req.Body).Decode(&pve); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	b := p.provider.PropertyValuesEqual(pve.Name, pve.Previous, pve.Current)
+	r := PropertyValuesEqualsResponse{
+		Equals: b,
+	}
+	out, err := json.Marshal(&r)
+	if err != nil {
+		log.Error(err)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(out)
+}
+
+func (p *HTTPProvider) adjustEndpointsHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		log.Errorf("Unsupported method %s", req.Method)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	pve := []*endpoint.Endpoint{}
+	if err := json.NewDecoder(req.Body).Decode(&pve); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	pve = p.provider.AdjustEndpoints(pve)
+	out, err := json.Marshal(&pve)
+	if err != nil {
+		log.Error(err)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(out)
+}
+
+func (p *HTTPProvider) negotiate(w http.ResponseWriter, req *http.Request) {
+	w.Header().Set(varyHeader, contentTypeHeader)
+	w.Header().Set(contentTypeHeader, mediaTypeFormatAndVersion)
+	w.WriteHeader(http.StatusOK)
+}
+
+// StartHTTPApi starts a HTTP server given any provider.
+// the function takes an optional channel as input which is used to signal that the server has started.
+// The server will listen on port 8888.
+// The server will respond to the following endpoints:
+// - /records (GET): returns the current records
+// - /records (POST): applies the changes
+// - /propertyvaluesequal (POST): executes the PropertyValuesEqual method
+// - /adjustendpoints (POST): executes the AdjustEndpoints method
```

Comment:
Where are the unit tests for this?

They should test starting on a port other than 8888.